test/common/run-tests

```diff
@@ -211,6 +213,9 @@ def run(opts, image):
     if len(changed_tests) > 3:
         changed_tests = []
 
+    # always mark check-no-retry as changed, it should only run once, as it's marked with @noretry
+    changed_tests.append("check-no-retry")
```

Comment:
I wouldn't do that -- *this* PR changes check-no-retry, so we can already see that it doesn't retry 3 times. And for other PRs that shouldn't matter either.